integration-test/src/test/java/com/datastrato/gravitino/integration/test/container/HiveContainer.java

```diff
@@ -83,7 +84,12 @@ protected boolean checkContainerStatus(int retryLimit) {
           isHiveContainerReady = true;
           break;
         }
-        Thread.sleep(5000);
+        LOG.info(
+            "Hive container is not ready, recheck({}/{}) after {}ms",
+            nRetry,
+            retryLimit,
+            sleepTimeMillis);
+        Thread.sleep(sleepTimeMillis);
```

Comment:
Is 5 seconds not enough to ensure that the Trino server has started?

Reply:
It's not a Trino server, it's HDFS initialization.

From the logs, it does seem that 5s is not enough. You can take a look at the description in the issue.

Reply:
I see.

Reply:
Can we use the exponent numbers like `1`, `2`, `4`, and so on, I believe it will function more well than fixed time.

Reply:
Taking 10 seconds to complete a check is too slow. 
If the service is normal in the second second, this program needs to wait for 10 seconds.
I suggest check the interface to:
···
 checkContainerStatus(int timeout)
···
Check every second, continuing until timeout or success.

Reply:
Frequent checking can result in additional performance overhead and stress on the target process. However, setting an overall timeout is a good suggestion. 

From the log information in the issue description, it is evident that HDFS completed initialization after 33 seconds of container startup, but the detection mechanism returned a failure at 28 seconds.

Considering that this method includes three checks (DataNode status, Hive connection, HDFS connection), I recommend setting the **`overall timeout`** to **`60s`** and increasing the retry interval exponentially. If the total time spent on retries exceeds the specified  **`overall timeout`**, it should be deemed a failed initialization. 

What do you think of this approach? @yuqi1129 @diqiu50